test/graph_output_tests.cpp

```diff
@@ -107,6 +107,181 @@ TEST_CASE("write_graph, enzymes, binary")
     }
 }
 
+TEST_CASE("write_graph, aves-songbird-social, binary")
+{
+    // First we read in a test graph
+    WeightedEdges32 edges;
+    auto emplace = [&](Vertex32 u, Vertex32 v, Weight w) { edges.push_back(WeightedEdge32{ u, Target32{ v, w } }); };
+    std::ifstream graph_input_unweighted_directed(graph_path + undirected_weighted_aves_songbird_social);
+
+    auto const [vertex_count, edge_count] =
+        read_graph<Vertex32, decltype(emplace), EdgeListUndirectedWeightedLoopStatic>(graph_input_unweighted_directed,
+                                                                                      std::move(emplace));
+
+    CHECK(vertex_count == aves_songbird_social_vertex_count);
+    CHECK(edge_count == aves_songbird_social_edge_count);
+    REQUIRE(edge_count == edges.size());
+
+    std::filesystem::path file_path = [&]() -> std::filesystem::path
+    {
+        if constexpr (developing_new_file_format)
+        {
+            return { graph_path + undirected_weighted_aves_songbird_social_bin };
+        }
+        else
+        {
+            return { graph_path + "test_graph.bin" };
+        }
+    }();
+
+    std::ofstream out_file = open_binary_file(file_path);
+
+    REQUIRE(out_file);
+
+    // Write Graph
+    write_graph<BinaryUndirectedWeightedStatic, Vertex32>(
+        out_file, edges, vertex_count, edge_count,
+        [](std::ofstream& o, WeightedEdge32 edge)
+        {
+            o.write(reinterpret_cast<const char*>(&edge.source), sizeof(Vertex32));
+            o.write(reinterpret_cast<const char*>(&edge.target.vertex), sizeof(Vertex32));
+            o.write(reinterpret_cast<const char*>(&edge.target.weight), sizeof(Weight));
+        });
+
+    // Now we read in the written graph and check if we read the expected data.
+    std::ifstream binary_graph(file_path);
+
+    BinaryGraphHeader header = read_binary_graph_header(binary_graph);
+    REQUIRE(header.vertex_id_byte_size == sizeof(Vertex32));
+    REQUIRE(header.weight_byte_size == sizeof(Weight));
+    REQUIRE(!header.directed);
+    REQUIRE(header.weighted);
+    REQUIRE(!header.dynamic);
+
+    WeightedEdges32 edges_in;
+    auto read_f = [&](std::ifstream& input)
+    {
+        edges_in.push_back(WeightedEdge32{});
+        input.read((char*)&edges_in.back().source, sizeof(Vertex32));
+        input.read((char*)&edges_in.back().target.vertex, sizeof(Vertex32));
+        input.read((char*)&edges_in.back().target.weight, sizeof(Weight));
+        return true;
+    };
+
+    auto [vertex_count_in, edge_count_in] = read_binary_graph(binary_graph, header, std::move(read_f));
+
+    // Note: EOF is an int (for most OS?)
```

Comment:
do you mean std::ios::eof()? Isn't it boolean?

Reply:
No, I am actually reading out the `EOF` symbol from the file. Sadly, I have not yet understood if when writing a file this symbol is always of type `int`. That's why I left he comment there.